python_scripts/linear_models.py

```diff
@@ -335,15 +341,18 @@ def f(x):
 
 # %% [markdown]
 # It is important to note that the model learnt will not be able to handle
-# the non-linearity linking `x` and `y` since it is beyond the assumption made
-# when using a linear model. To obtain a better model, we have mainly 3
-# solutions: (i) choose a model that natively can deal with non-linearity,
-# (ii) "augment" features by including expert knowledge which can be used by
-# the model, or (iii) use a "kernel" to have a locally-based decision function
-# instead of a global linear decision function.
+# the non-linear relationship between `x` and `y` since linear models assume
+# the relationship between `x` and `y` to be linear. To obtain a better model,
+# we have 3 main solutions:
+#
+# 1. choose a model that natively can deal with non-linearity,
+# 2. "augment" features by including expert knowledge which can be used by
+#    the model, or
+# 2. use a "kernel" to have a locally-based decision function instead of a
+#    global linear decision function.
```

Comment:
This is something on which I am not settled. We might skip the kernel here because it is too advanced.
It would be presented in the feature engineering notebook.
I put it here because it was natural but I think that we need to consider removing/moving it or add in simple terms what is this (a kernel will make a transform of the original dataset to find relationships).